tools/render-test/render-d3d11.cpp

```diff
@@ -612,165 +578,149 @@ void* D3D11Renderer::map(ID3D11Buffer * buffer, MapFlavor flavor)
     // We update our constant buffer per-frame, just for the purposes
     // of the example, but we don't actually load different data
     // per-frame (we always use an identity projection).
-    D3D11_MAPPED_SUBRESOURCE dxMapped;
-    HRESULT hr = dxContext->Map(dxBuffer, 0, dxMapFlavor, 0, &dxMapped);
+    D3D11_MAPPED_SUBRESOURCE mappedSub;
+    HRESULT hr = m_immediateContext->Map(buffer, 0, mapType, 0, &mappedSub);
     if (FAILED(hr))
         return nullptr;
 
-    return dxMapped.pData;
+    return mappedSub.pData;
 }
 
 void* D3D11Renderer::map(Buffer* buffer, MapFlavor flavor)
 {
     return map(((BufferImpl*)buffer)->buffer, flavor);
 }
 
-void D3D11Renderer::unmap(ID3D11Buffer * buffer)
+void D3D11Renderer::unmap(ID3D11Buffer* buffer)
 {
-    auto dxContext = dxImmediateContext;
-    dxContext->Unmap(buffer, 0);
+    m_immediateContext->Unmap(buffer, 0);
 }
 
-void D3D11Renderer::unmap(Buffer* buffer)
+void D3D11Renderer::unmap(Buffer* bufferIn)
 {
-    auto dxBuffer = ((BufferImpl*)buffer)->buffer;
-    unmap(dxBuffer);
+    unmap(((BufferImpl*)bufferIn)->buffer);
 }
 
-void D3D11Renderer::setInputLayout(InputLayout* inputLayout)
+void D3D11Renderer::setInputLayout(InputLayout* inputLayoutIn)
 {
-    auto dxContext = dxImmediateContext;
```

Comment:
To be clear, a lot of these lines that initialize `dxContext` were doing two things:

1) Making the rest of the function body be parameterized on the choice of context, in case we ever swapped in a different context (yes, I know we are unlikely to use D3D11 deferred command lists at this point). We could switch to use a different context just by changing this one line, instead of every use site.

2) It also avoids indirecting through `this` at ever use site, since there is now a single load from `this` to a local variable, instead of (potentially) one load per use site if the C++ compiler can't resolve all the aliasing.